includes/Core/Authentication/Authentication.php

```diff
@@ -586,6 +586,21 @@ private function inline_js_admin_data( $data ) {
 			$data['userData']['picture'] = $profile_data['photo'];
 		}
 
+		add_filter(
+			'googlesitekit_user_data',
+			function( $user ) use ( $data ) {
+				$current_user = wp_get_current_user();
+				$new_data     = array(
+					'id'      => $current_user->ID,
+					'email'   => $data['userData']['email'],
+					'name'    => $current_user->display_name,
+					'picture' => $data['userData']['picture'],
+				);
+				$user['user'] = $new_data;
+				return $user;
+			}
+		);
```

Comment:
Let's decouple this filter hook from `inline_js_admin_data` (since that is legacy) and hook it in in the `register` method. The logic here regarding getting the (Google account) profile data can be duplicated, that's fine in this case (legacy vs non-legacy). We also don't need to use `wp_get_current_user`, let's only update the parts of the `$user['user']` data that need to be updated (`email` and `picture`, if set).